deepchem/models/torch_models/tests/test_hf_models.py

```diff
@@ -0,0 +1,194 @@
+    pass
+
+
+@pytest.fixture
+def smiles_dataset(tmpdir):
+    import deepchem as dc
+    import pandas as pd
+    smiles = [
+        "CCN(CCSC)C(=O)N[C@@](C)(CC)C(F)(F)F",
+        "CC1(C)CN(C(=O)Nc2cc3ccccc3nn2)C[C@@]2(CCOC2)O1"
+    ]
+    labels = [3.112, 2.432]
+    df = pd.DataFrame(list(zip(smiles, labels)), columns=["smiles", "task1"])
+    filepath = os.path.join(tmpdir, 'smiles.csv')
+    df.to_csv(filepath)
+
+    loader = dc.data.CSVLoader(["task1"],
+                               feature_field="smiles",
+                               featurizer=dc.feat.DummyFeaturizer())
+    dataset = loader.create_dataset(filepath)
+    return dataset
+
+
+@pytest.fixture
+def hf_tokenizer(tmpdir):
+    filepath = os.path.join(tmpdir, 'smiles.txt')
+    with open(filepath, 'w') as f:
+        f.write(
+            'CN(c1ccccc1)c1ccccc1C(=O)NCC1(O)CCOCC1\nCC[NH+](CC)C1CCC([NH2+]C2CC2)(C(=O)[O-])C1\n'
+        )
+        f.write(
+            'COCC(CNC(=O)c1ccc2c(c1)NC(=O)C2)OC\nOCCn1cc(CNc2cccc3c2CCCC3)nn1\n'
+        )
+        f.write(
+            'CCCCCCc1ccc(C#Cc2ccc(C#CC3=CC=C(CCC)CC3)c(C3CCCCC3)c2)c(F)c1\nO=C(NCc1ccc(F)cc1)N1CC=C(c2c[nH]c3ccccc23)CC1\n'
+        )
+    from tokenizers import ByteLevelBPETokenizer
+    from transformers.models.roberta import RobertaTokenizerFast
+    tokenizer = ByteLevelBPETokenizer()
+    tokenizer.train(files=filepath,
+                    vocab_size=1_000,
+                    min_frequency=2,
+                    special_tokens=["<s>", "<pad>", "</s>", "<unk>", "<mask>"])
+    tokenizer_path = os.path.join(tmpdir, 'tokenizer')
+    os.makedirs(tokenizer_path)
+    tokenizer.save_model(tokenizer_path)
+    tokenizer = RobertaTokenizerFast.from_pretrained(tokenizer_path)
+    return tokenizer
+
+
+@pytest.mark.torch
+def test_pretraining(hf_tokenizer, smiles_dataset):
+    from deepchem.models.torch_models.hf_models import HuggingFaceModel
+    from transformers.models.roberta import RobertaConfig, RobertaForMaskedLM
+
+    config = RobertaConfig(vocab_size=hf_tokenizer.vocab_size)
+    model = RobertaForMaskedLM(config)
+
+    hf_model = HuggingFaceModel(model=model,
+                                tokenizer=hf_tokenizer,
+                                task='pretraining')
+    loss = hf_model.fit(smiles_dataset, nb_epoch=1)
+
+    assert loss
+
+
+@pytest.mark.torch
+def test_hf_model_regression(hf_tokenizer, smiles_dataset):
+    from transformers.models.roberta import (RobertaConfig,
+                                             RobertaForSequenceClassification)
+
+    config = RobertaConfig(vocab_size=hf_tokenizer.vocab_size,
+                           problem_type='regression',
+                           num_labels=1)
+    model = RobertaForSequenceClassification(config)
+    hf_model = HuggingFaceModel(model=model,
+                                tokenizer=hf_tokenizer,
+                                task='finetuning')
+    hf_model.fit(smiles_dataset, nb_epoch=1)
```

Comment:
The fact that tokenization is done on the fly in the HuggingFaceModel fit should be documented and discussed in the main docstring since it's a major change from the way DeepChem usually operates. It's roughly analogous to FitTransformer (?)

Reply:
added to main docstring